core/domain/state_domain.py

```diff
@@ -745,16 +758,336 @@ def get_all_html_content_strings(self):
+            interaction_id).customization_arg_specs
+        customization_args = interaction_dict['customization_args']
+        for ca_spec in ca_specs:
+            ca_spec_name = ca_spec.name
+            if ca_spec_name in customization_args:
+                customization_args[ca_spec_name]['value'] = (
+                    traverse_ca_by_schema_and_convert_html(
+                        customization_args[ca_spec_name]['value'],
+                        ca_spec.to_dict()['schema']
+                    )
+                )
+
+        return interaction_dict
+
+    @staticmethod
+    def convert_customization_args_dict_to_customization_args(
+            interaction_id, customization_args_dict):
+        """Converts customization arguments dictionary to customization
+        arguments. This is done by converting each customization argument to a
+        InteractionCustomizationArg domain object.
+
+        Args:
+            interaction_id: str. The interaction id.
+            customization_args_dict: dict. A dictionary of customization
+                argument name to a customization argument dict, which is a dict
+                of the single key 'value' to the value of the customization
+                argument.
+
+        Returns:
+            dict. A dictionary of customization argument names to the
+            InteractionCustomizationArg domain object's.
+        """
+        if interaction_id is None:
+            return {}
+
+        customization_args = {}
+        ca_specs = interaction_registry.Registry.get_interaction_by_id(
+            interaction_id).customization_arg_specs
+
+        for ca_spec in ca_specs:
+            ca_name = ca_spec.name
+            customization_args[ca_name] = (
+                InteractionCustomizationArg.from_customization_arg_dict(
+                    customization_args_dict[ca_name],
+                    ca_spec.schema
+                )
+            )
+
+        return customization_args
+
+
+class InteractionCustomizationArg(python_utils.OBJECT):
+    """Object representing an interaction's customization argument.
+    Any SubtitledHtml or SubtitledUnicode values in the customization argument
+    value are represented as their respective domain objects here, rather than a
+    SubtitledHtml dict or SubtitledUnicode dict.
+    """
+
+    def to_customization_arg_dict(self):
+        """Converts a InteractionCustomizationArgument domain object to a
+        a customization argument dictionary. This is done by
+        traversing the customization argument schema, and converting
+        SubtitledUnicode to unicode and SubtitledHtml to html where appropriate.
+        """
+        def convert_content_to_dict(ca_value, unused_obj_type):
+            """Conversion function used to convert SubtitledHtml to
+            SubtitledHtml dicts and SubtitledUnicode to SubtitledUnicode dicts.
+
+            Args:
+                ca_value: dict. Dictionary of key 'value' to
```

Comment:
This could be clearer. E.g. "A dictionary with one key, 'value', whose corresponding value is ..."